prometheus/client.go

```diff
@@ -33,40 +31,72 @@ func NewClient() (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	client.p8s = p8s
+	client.api = v1.NewAPI(p8s)
 	return &client, nil
 }
 
+// For testing / mocking
+func (in *Client) inject(api v1.API) {
+	in.api = api
+}
+
 // GetSourceServices returns a map of source services for a given service identified by its namespace and service name.
 // Returned map has a destination version as a key and a "<origin service>/<origin version>" pair as value.
 // Destination service is not included in the map as it is passed as argument.
 // It returns an error on any problem.
-func (in *Client) GetSourceServices(namespace string, servicename string) (map[string]string, error) {
-	query := fmt.Sprintf(istioRequestQuery, servicename, namespace)
-	api := v1.NewAPI(in.p8s)
-	result, err := api.Query(context.Background(), query, time.Now())
+func (in *Client) GetSourceServices(namespace string, servicename string) (map[string][]string, error) {
```

Comment:
Please, update the comments describing the signature as return type has changed from map[string]string to map[string][]string so here should be changed:
https://github.com/swift-sunshine/swscore/pull/30/files#diff-970c77c7471b416af8e59de92b0accc0R44